lib/fluent/registry.rb

```diff
@@ -43,11 +46,18 @@ def lookup(type)
       raise ConfigError, "Unknown #{@kind} plugin '#{type}'. Run 'gem search -rd fluentd-plugin' to find plugins"  # TODO error class
     end
 
+    def reverse_lookup(value)
+      @map.each do |k, v|
+        return k if v == value
+      end
+      nil
+    end
+
     def search(type)
       path = "#{@search_prefix}#{type}"
 
       # prefer LOAD_PATH than gems
-      files = $LOAD_PATH.map { |lp|
+      files = ($LOAD_PATH + @paths).map { |lp|
```

Comment:
Ah, this code sorts found list and take last member of it to get latest version of files (it has another incorrectness, but it's out of scope for now).
https://github.com/fluent/fluentd/pull/838/files#diff-da556f72b0de83b0e659240bf9d879bbR66

So we should check files separately for `@paths` and `$LOAD_PATH`.